Replace child_process with cross-spawn for cross-platform compatibility

ahutchings opened the pull request:
Fixes #10

ahutchings commented:
@kruppel Would you mind pulling this in and cutting a patch release when you get a chance?

kruppel commented on package.json:
```diff
@@ -14,6 +14,7 @@
   "license": "Apache-2.0",
   "repository": "http://github.com/kruppel/check-engines",
   "dependencies": {
+    "cross-spawn": "^5.0.1",
```
@ahutchings I'm a little hesitant to pull in a dependency for this, though it's tricky if the `spawn` option works for only Node versions >= 6. Beyond that, should the version range be so strict? Is there any reason to not set it to `>=5.0.1`?

ahutchings replied:
In addition to the shell support on node < v6 issue you mentioned, I found that we would need to detect the platform and conditionally run "npm" or "npm.cmd". cross-spawn does even better by supporting `PATHEXT`. All in all it felt like I would be re-engineering a solution that was already solved by the dependency, so I feel that it's worthwhile to pull it in given that it requires so little code change.

Regarding the version range - the carat is generally safer than `>=` because it won't pull in a version of cross-spawn that has breaking changes.

kruppel commented on lib/engine-test.js:
```diff
@@ -52,7 +52,7 @@ EngineTest.prototype.getVersion = function(callback) {
   var result = '';
   var flag = ENGINE_TEST_ARGS[this.type] || '-v';
-  var version = childProcess.spawn(this.type, [flag]);
+  var version = crossSpawn.spawn(this.type, [flag]);
```
In `5.0.0`, [work was done to support the `shell` option](https://github.com/IndigoUnited/node-cross-spawn/pull/46). Should that be passed as an option here?

ahutchings replied:
Looks like cross-spawn will attempt to use the native child_process "shell" option if the flag is passed and the node version supports it. I don't have a Windows machine with me at the moment but I'll try it out this evening.

ahutchings commented:
I updated the spawn call to pass `{shell: true}`. Using this option is more forward-looking, as it would be needed if we wanted to drop node < 6 support in the future. Passing the shell option also allows cross-spawn to use native node child_process.spawn support and avoid unnecessary parsing of the command.

I also added node 7 to the list of node versions to test, to ensure everything works as expected there. I can revert that change if you like.

It would be nice to add automatic Windows testing against multiple node versions using something like [Appveyor](https://www.appveyor.com/docs/lang/nodejs-iojs/). I haven't used Appveyor before but I'll do what I can to help configure it if you want to add it.

kruppel commented:
@ahutchings v1.5.0 released. Thanks for the contribution!
